ambry-store/src/test/java/com.github.ambry.store/CompactionManagerTest.java

```diff
@@ -0,0 +1,248 @@
+   */
+  @Test
+  public void testDifferentThresholdsForMinLogSizeToCompact() throws StoreException {
+    int[] minLogSizeToTriggerCompactionInPercentages = new int[]{10, 20, 35, 40, 50, 59, 60, 61, 65, 70, 80, 95};
+    // when used capacity(60%) is <= (minLogSize) % of total capacity, compactionDetails is expected to be null.
+    // If not, validLogSegments needs to be returned
+    for (int minLogSize : minLogSizeToTriggerCompactionInPercentages) {
+      properties.setProperty("store.min.log.size.to.trigger.compaction.in.percent", String.valueOf(minLogSize));
+      config = new StoreConfig(new VerifiableProperties(properties));
+      MetricRegistry metricRegistry = new MetricRegistry();
+      StorageManagerMetrics metrics = new StorageManagerMetrics(metricRegistry);
+      blobStore = new MockBlobStore(config, metrics, time, CAPACITY_IN_BYTES, DEFAULT_USED_CAPACITY_IN_BYTES);
+      compactionManager = new CompactionManager(config, time);
+      blobStore.validLogSegments = generateRandomStrings(2);
+      if (blobStore.usedCapacity <= (
+          config.storeMinUsedCapacityToTriggerCompactionInPercentage * blobStore.capacityInBytes / 100)) {
+        verifyCompactionDetails(null);
+      } else {
+        verifyCompactionDetails(
+            new CompactionDetails(time.milliseconds() - messageRetentionTimeInMs, blobStore.validLogSegments));
+      }
+    }
+  }
+
+  /**
+   * Tests {@link CompactionManager#getCompactionDetails(BlobStore)} for different values for
+   * {@link StoreConfig#storeDeletedMessageRetentionDays}
+   */
+  @Test
+  public void testDifferentMessageRetentionDays() throws StoreException, InterruptedException {
+    int[] messageRetentionDayValues = new int[]{1, 2, 3, 6, 9};
+    time.sleep(10 * messageRetentionTimeInMs);
+    for (int messageRetentionDays : messageRetentionDayValues) {
+      properties.setProperty("store.deleted.message.retention.days", String.valueOf(messageRetentionDays));
+      config = new StoreConfig(new VerifiableProperties(properties));
+      MetricRegistry metricRegistry = new MetricRegistry();
+      StorageManagerMetrics metrics = new StorageManagerMetrics(metricRegistry);
+      blobStore = new MockBlobStore(config, metrics, time, CAPACITY_IN_BYTES, DEFAULT_USED_CAPACITY_IN_BYTES);
+      compactionManager = new CompactionManager(config, time);
+      blobStore.validLogSegments = generateRandomStrings(2);
+      verifyCompactionDetails(
+          new CompactionDetails(time.milliseconds() - messageRetentionDays * Time.SecsPerDay * Time.MsPerSec,
+              blobStore.validLogSegments));
+    }
+  }
+
+  /**
+   * Test {@link CompactionManager#getCompactionDetails(BlobStore)} for failure cases
+   * @throws StoreException
+   */
+  @Test
+  public void testConstructionFailureTest() throws StoreException {
+    blobStore.validLogSegments = Collections.EMPTY_LIST;
+    try {
+      compactionManager.getCompactionDetails(blobStore);
```

Comment:
this isn't an object construction failure - could you name the test appropriately?